resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerAspect.java

```diff
@@ -127,6 +127,41 @@ private Object defaultWebFlux(ProceedingJoinPoint proceedingJoinPoint, io.github
 		}
 	}
 
+	/**
+	 * handle the CompletionStage return types AOP based into configured circuit-breaker
+	 */
+	private Object defaultCompletionStage(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker, String methodName) throws Throwable {
```

Comment:
Please have a look at the implementation of `CircuitBreaker.decorateCompletionStage`.
We can copy it to CircuitBreakerAspect or we have to support a `CheckedFunction0<CompletionStage<?>>` 
But I don't like the signature :(

```
CheckedFunction0<CompletionStage<?>> stageSupplier = () -> (CompletionStage<?>) proceedingJoinPoint.proceed();
return circuitBreaker.executeCompletionStage(stageSupplier);
```